galaxy_ng/app/api/v1/tasks.py

```diff
@@ -0,0 +1,254 @@
+import copy
+import datetime
+import logging
+import os
+import tempfile
+
+from galaxy_importer.loaders.content import RoleLoader
+from galaxy_importer.utils import markup as markup_utils
+
+from galaxy_ng.app.utils.galaxy import upstream_role_iterator
+from galaxy_ng.app.utils.git import get_tag_commit_hash
+from galaxy_ng.app.utils.git import get_tag_commit_date
+import galaxy_ng.app.utils.roles as roles_utils
+
+from galaxy_ng.app.api.v1.models import LegacyNamespace
+from galaxy_ng.app.api.v1.models import LegacyRole
+
+from git import Repo
+
+
+logger = logging.getLogger(__name__)
+
+
+def legacy_role_import(
```

Comment:
@awcrosby can galaxy_importer handle role imports?

Reply:
galaxy_importer only handles collections... it has pieces that process a role inside a collection, but not a standalone role.

This method does use some the utils from galaxy_importer to process the role readme.